lib/services/fetch_product.rb

```diff
@@ -0,0 +1,10 @@
+module Shop
+  class FetchProduct
+    def call(id)
+      return unless id
+      PRODUCTS.find do |prod|
```

Comment:
you can use one-liner in such cases: `PRODUCTS.find { |product| product.id == id.to_i }`